crates/sui-framework/src/natives/crypto/ed25519.rs

```diff
@@ -1,37 +1,70 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
-use crate::legacy_empty_cost;
+use crate::natives::NativesCostTable;
 use fastcrypto::{
     ed25519::{Ed25519PublicKey, Ed25519Signature},
     traits::{ToFromBytes, VerifyingKey},
 };
 use move_binary_format::errors::PartialVMResult;
-use move_vm_runtime::native_functions::NativeContext;
+use move_core_types::gas_algebra::InternalGas;
+use move_vm_runtime::{native_charge_gas_early_exit, native_functions::NativeContext};
 use move_vm_types::{
     loaded_data::runtime_types::Type,
     natives::function::NativeResult,
     pop_arg,
     values::{Value, VectorRef},
 };
 use smallvec::smallvec;
-use std::collections::VecDeque;
+use std::{collections::VecDeque, ops::Mul};
 
+#[derive(Clone)]
+pub struct Ed25519VerifyCostParams {
+    /// Base cost for invoking the `ed25519_verify` function
+    pub ed25519_ed25519_verify_cost_base: InternalGas,
+    /// Cost per byte of `msg`
+    pub ed25519_ed25519_verify_msg_cost_per_byte: InternalGas,
```

Comment:
Same with the discussion we had for hash functions, we are in practice interested in the block size (not per byte) of the internal hash function of ed25519, which is SHA512 (block size for sha512 is 1024bytes)

Reply:
Block size is 128 here

Reply:
addressed now.